tests/test.c

```diff
@@ -647,3 +647,27 @@ void check_param_range(char* dev_name, uint64_t uid, char* param_name, param_typ
     }
     print_pass();
 }
+
+// Returns if the latency is neither too high nor negative
+void check_latency(char* dev_name, uint64_t uid, char* param_name, int32_t upper_bound_latency, int32_t start_time) {
+    uint8_t dev_type = device_name_to_type(dev_name);
+    device_t* dev = get_device(dev_type);
+
+    param_val_t vals_after[dev->num_params];
+    uint32_t param_idx = (uint32_t) get_param_idx(dev_type, param_name);
+    device_read_uid(uid, EXECUTOR, DATA, (1 << param_idx), vals_after);
+    param_val_t end_time = vals_after[param_idx];
+    int32_t elapsed_time = end_time.p_i - start_time;
```

Comment:
This function only seems useful if its a `TimeTestDevice`. Is there any point to requiring a `dev_name` and `param_name` then?

Reply:
This was there just in case in the future someone decides to write a new virtual device with a `TIMESTAMP` on it. I do agree though, the `param_name` is kind of redundant

Reply:
Yea this is interesting and didn't comment on this because we may want future devices with `TIMESTAMP` params. Though I'm having trouble thinking of a scenario when we would need a new device.

Reply:
Yea thats a good point, they can easily just connect the `TimeTestDevice`. Its an easy fix to the function if they do write a new device in the future. I'll remove it from there